pika/adapters/blocking_connection.py

```diff
@@ -700,6 +700,18 @@ def channel(self, channel_number=None):
 
         return channel
 
+    def __enter__(self):
+        # Prepare `with` context
+        return self
+
+    def __exit__(self, tp, value, traceback):
+        # Close connection after `with` context
+        if isinstance(value, (KeyboardInterrupt, SystemExit, GeneratorExit)):
```

Comment:
@reddec: on second thought, let's keep them all the same; let's just call `self.cose()` in all cases. Sorry for the initial feedback - I was thinking out loud.